### PR TITLE
simplify some logic

### PR DESCRIPTION
**Pull request by wuyudi**
## Description

1. simplify some logic

Fixes issue #

## TODOs:

- [x] Changes to fish usage are reflected in user documentation/manpages.
- [x] Tests have been added for regressions fixed
- [x] User-visible changes noted in CHANGELOG.rst

### CODE REVIEW

File: fish-rust/src/path.rs [hunk 1]
```diff
@@ -180,11 +180,7 @@ fn maybe_issue_path_warning(
 /// \returns the path if found, none if not.
 pub fn path_get_path(cmd: &wstr, vars: &dyn Environment) -> Option<WString> {
     let result = path_try_get_path(cmd, vars);
-    if result.err.is_some() {
-        None
-    } else {
-        Some(result.path)
-    }
+    result.err.map_or(Some(result.path), |_| None)
```
Comment (Xiretza) [inline fish-rust/src/path.rs:183]: I honestly don't think this makes it clearer at all.

File: fish-rust/src/common.rs [hunk 2]
```diff
@@ -1464,15 +1455,16 @@ pub fn write_loop<Fd: AsRawFd>(fd: &Fd, buf: &[u8]) -> std::io::Result<usize> {
 pub fn read_loop<Fd: AsRawFd>(fd: &Fd, buf: &mut [u8]) -> std::io::Result<usize> {
```
Comment (mqudsi) [inline fish-rust/src/common.rs:1455]: `read_loop()` is going to be thrown away soon.

File: fish-rust/src/threads.rs [hunk 3]
```diff
@@ -20,7 +21,7 @@ impl FloggableDebug for ThreadId {}
 // efficient/fast/low-overhead code.
 
 /// The thread id of the main thread, as set by [`init()`] at startup.
-static mut MAIN_THREAD_ID: Option<ThreadId> = None;
+static MAIN_THREAD_ID: OnceCell<std::thread::ThreadId> = OnceCell::new();
```
Comment (mqudsi) [inline fish-rust/src/threads.rs:24]: No thanks. We're intentionally using a lock-free option.

**Review by mqudsi — CHANGES_REQUESTED** [review]
I'm happy with the `MAIN_THREAD_ID` code as it was. It's intentionally lock-free and as cheap as can be.

**Pull request merged**